src/libraries/src.proj

```diff
@@ -0,0 +1,38 @@
+<Project Sdk="Microsoft.Build.Traversal">
+
+  <PropertyGroup>
+    <TraversalGlobalProperties>BuildAllProjects=true</TraversalGlobalProperties>
+  </PropertyGroup>
+
+  <ItemGroup>
+    <NativeBinPlaceProject Include="Native\native-binplace.proj" />
+    <ManualShimProject Include="shims\manual\*.csproj" />
+  </ItemGroup>
+
+  <ItemGroup>
+    <!-- Restore only and build before. -->
+    <ProjectReference Include="@(NativeBinPlaceProject)" Condition="'$(MSBuildRestoreSessionId)' != ''" />
+    <ProjectReference Include="$(MSBuildThisFileDirectory)*\src\*.csproj" Exclude="@(ProjectExclusions)" />
+    <ProjectReference Include="$(MSBuildThisFileDirectory)*\src\*.ilproj" Exclude="@(ProjectExclusions)" />
+    <ProjectReference Include="$(MSBuildThisFileDirectory)*\src\*.vbproj" Exclude="@(ProjectExclusions)" />
+    <!-- Explicitly include the runtime.depproj project here to correctly set up the testhost. -->
+    <ProjectReference Include="restore\runtime\runtime.depproj" />
+    <!-- Restore only and build later. -->
+    <ProjectReference Include="@(ManualShimProject)" Condition="'$(MSBuildRestoreSessionId)' != ''" />
+  </ItemGroup>
+
+  <Target Name="NativeBinPlace"
+          BeforeTargets="Build">
```

Comment:
Should we instead hook these targets up via `BuildDependsOn`? We try to avoid `Before/AfterTargets` as they are prone to not flowing errors correctly.

Reply:
? I'm hearing of that the first time. Can you elaborate or point me to an issue?

Reply:
I don't recall the exact issue, @joperezr has mentioned that before, so I'll let him respond.